src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -106,6 +110,14 @@ object DepositHandler {
           _ <- props.setState(INVALID, msg)
           _ <- props.save()
         } yield ()
+      case RejectedDepositException(_, msg, cause) =>
+        log.error(s"[$id] Rejected deposit", cause)
+        for {
+          _ <- removeZipFiles(depositDir) // Currently, the only reason for SWORD 2 to reject a deposit, is insufficient disk space, so we clean up, here.
```

Comment:
I want to keep the exceptions aligned with the external states, and and the same time limit those external states to a relatively small and well-defined number. Both `FAILED` and `REJECTED` make sense, here. `FAILED` because the problem lies on the server, and `REJECTED` because we actually reject the deposit for being too large to process. Also, we  decided earlier to only show a generic message to the client in case of `FAILED` deposits, but I think that in this case it makes more sense to actually tell the client what the problem is, so that they may refrain from sending more large (but in fact valid) deposits.